frontend/ui/network/manager.lua

```diff
@@ -139,6 +139,9 @@ function NetworkMgr:turnOnWifiAndWaitForConnection(callback)
     self:scheduleConnectivityCheck(callback, info)
 end
 
+--- @note: The callback will only run *after* a *succesful* network connection.
+---        The only guarantee it provides is isConnected (i.e., an IP & a local gateway),
+---        *NOT* isOnline (i.e., WAN), se be careful with recursive callbacks!
 function NetworkMgr:beforeWifiAction(callback)
```

Comment:
A bit hard to grasp these kind of constructs (and a bit ugly, all with a @note):
```lua
    if not NetworkMgr:isOnline() then
        --- @note: Avoid infinite recursion, beforeWifiAction only guarantees isConnected, not isOnline.
        if not NetworkMgr:isConnected() then
            NetworkMgr:beforeWifiAction(function() self:lookupWikipedia(word, box, get_fullpage, forced_lang) end)
        else
            NetworkMgr:beforeWifiAction()
        end
```
Can't we passe a 2nd parameter to `beforeWifiAction(callback, wants_online=true/false)` (so it guarantees what we want) and have that shorter and more obvious ?
(just asking, I'm not even familiar with beforeWifiAction & co)

Reply:
That crossed my mind, but I kind of like those constructs, they make it extremely obvious whether the action requires isOnline or not (since lua doesn't have named parameters).

That said, it's not particularly pretty, so I might come up with a helper function that wraps the whole thing and has  a potentially more meaningful name than beforeWifiAction, though ;p